### ISSUE TITLE
Two 404 .html files will be created on build

### ISSUE THREAD

**Issue by konstantinschuette** [gatsbyjs/gatsby#36314]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I found out that when I run lighthouse-ci and have the auto page (html file) detection turned on, two .html files for the 404 site will be tested.

Because with a fresh blank Gatsby project you generate the following .html files on build:
`.public/index.html`
`.public/404.html`
`.public/404/index.html`

### Reproduction Link

https://githubbox.com/gatsbyjs/gatsby-starter-minimal

### Steps to Reproduce

1. Download the gatsby-starter-minimal or just run `gatsby new`
2. `gatsby build`
3. Look at the generated files

### Expected Result

Only one html file should be created on build for the 404 page.

### Actual Result

Two html files are generated for the 404 page.
![image](https://user-images.githubusercontent.com/18708105/182660985-28c05547-70b2-4d9b-88ff-bb0fa9a436a9.png)

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.4
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.77)
  npmPackages:
    gatsby: ^4.20.0 => 4.20.0
  npmGlobalPackages:
    gatsby-cli: 4.19.0
```

**Comment by JayboBaker** [comment]
Has this been fixed? I'm seeing the same issue, even with the latest version of `4.21.0`

**Comment by konstantinschuette** [comment]
> Has this been fixed? I'm seeing the same issue, even with the latest version of `4.21.0`

Hey, I should have linked the "new" issue that targets this problem correctly.
TLDR: It is intended that two 404 files are created.

More info: https://github.com/gatsbyjs/gatsby/discussions/36317#discussioncomment-3321799
